### DISCUSSION TITLE
App purpose

### DISCUSSION THREAD

**Discussion by mathe-matician** [jazzyclimber/beej-zach-app#4] [General]
Most of our talks about the app have been centered around problems within the dance community. So it has mostly been my understanding that we would focus on dance community specific implementations (e.g. possibly hard coding anything related to dance, pulling data from dance venues/events). I don't necessarily think the problems they face are unique to that community, though. So, I've changed my way I've been thinking about it to be more general, thinking of the app as an event/venue scheduling app. It seems it is more that existing event/venue apps just aren't targeting the niche dance market or the dance community is fine just using their current scheduling methods. 

I think keeping things generalized would help in two main ways:
1. A great learning opportunity to focus on abstraction and flexibility 
2. In the long run if we ever wanted to support different "markets", we could

I'm just checking that this is/was your thought process as well.

**Comment by jazzyclimber** [comment]
Yes -- On the whole, I agree.

I do think that event registration as a whole is an issue faced across the spectrum. 

However, certain portions of the app fall apart a bit if we make it generic. At least until a dominant market share is gained where folks buy-in to the app.

Maybe not actually. It's really just perhaps designed for a specific kind of event that maybe has a schedule. 

Regardless. I know that there are a handful of competitors in the market when it comes to Generalized event registration. 

With the swing dance community in particular I know for a fact that there is a vacuum. And I think that all it would take for this to catch on there is one successful event where we can prove that it solves a pain point for those event organizers. And if it's easy to use for a dancer and streamlines their registration process it will def take on there. 

I think the difficult part making it a generic thing would be getting folks to be vested in using it. Let alone trying to get venues to use it when they have things that "are working" for them already. 

------

It might be easier to build it out as a dance platform first and then if it works out using that experience as a "beta" if you will for something that would be more "global". The business person in me says that it's better to target a specific market than create something that could be used across the board. Those "global" products tend to be relatively generic and harder to sell. 

I think the point im trying to make is that I know there is a need in the swing dance community and I am fairly certain that if done correctly this will be a hit there. 

Does that make sense? I feel like im not explaining myself very well.

**Reply by mathe-matician** [reply]
Yep, your concerns about keeping the purpose focused make sense and I mostly agree with. Part of my argument is to not pigeon-hole ourselves with code design choices that are too specific as they could easily get too integrated making any possible refactoring in the future a real pain. That focus on generic code design doesn't mean we can't focus on the dance market, though. Ultimately, it is more of a "keep this in mind and err on generic" when reviewing, but not necessarily a "everything must be generic" thing.

I think most of that should come naturally in the code, where refactoring would rely on just adding new fields and update any size restrictions/checks that may be in place.

**Reply by jazzyclimber** [reply]
For sure -- I def agree with you here. I think it would make sense -- before we build a feature to talk about perhaps how we intend to use it initially (for the dance version) and how we might want to use it for the generic version. This way we can build the foundation for that plumbing right away or at least know that we can plug into it later on.

But im def all about making this easy to modulate on in the future. So planning things out best that we can makes a lot of sense.
